synapse/federation/transport/client.py

```diff
@@ -1192,10 +1192,24 @@ async def get_space_summary(
         )
 
     async def get_room_hierarchy(
-        self,
-        destination: str,
-        room_id: str,
-        suggested_only: bool,
+        self, destination: str, room_id: str, suggested_only: bool
+    ) -> JsonDict:
```

Comment:
We could choose whether to v1 vs. unstable based on a parameter but the other bits in this file seem to have separate methods for v1 vs. v2, etc. so I followed that pattern. I think we can clean that up separately if it makes sense?